site/jekyll/getting-started/tutorial_aspnet4.md

```diff
@@ -961,44 +981,45 @@ If you go to this URL in your browser, you should notice that the code has been
 
 ## Optimization: Server-side rendering
 
-Server-side rendering means that your application initially renders the components on the server-side, rather than fetching data from the server and rendering using JavaScript. This enhances the performance of your application since the user will see the initial state immediately.
+Server-side rendering means that your application initially renders the components on the server-side, rather than fetching data from the server and rendering using the client. Server-side rendering enhances the performance of your application since the user will see the initial state immediately.
 
 We need to make some modifications to `CommentBox` to support server-side rendering. Firstly, we need to accept an `initialData` prop, which will be used to set the initial state of the component, rather than doing an AJAX request. We also need to remove the `loadCommentsFromServer` call from `componentDidMount`, since it is no longer required. Also, we need to remove the `ReactDOM.render` call from the JSX file, as server-side rendering automatically outputs the correct `ReactDOM.render` call for you.
 
-```javascript{28}
-var CommentBox = React.createClass({
-  loadCommentsFromServer: function() {
-    var xhr = new XMLHttpRequest();
+```javascript{4,31-33}
+class CommentBox extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { data: this.props.initialData };
+		this.handleCommentSubmit = this.handleCommentSubmit.bind(this);
+  }
+  loadCommentsFromServer() {
+    const xhr = new XMLHttpRequest();
     xhr.open('get', this.props.url, true);
-    xhr.onload = function() {
-      var data = JSON.parse(xhr.responseText);
+    xhr.onload = () => {
+      const data = JSON.parse(xhr.responseText);
       this.setState({ data: data });
-    }.bind(this);
+    };
     xhr.send();
-  },
-  handleCommentSubmit: function(comment) {
-    var comments = this.state.data;
-    var newComments = comments.concat([comment]);
+  }
+	handleCommentSubmit(comment) {
+    const comments = this.state.data;
+    comment.Id = comments.length + 1;
```

Comment:
This looks like a bug fix, was it intentional?

Reply:
Yeah. React DevTools was yelling at me about using a key, and I couldn't figure out what was wrong because you were already using a key. Then I saw the property difference and realized what was happening.